tools/run_tests/performance/loadtest_template.py

```diff
@@ -81,42 +101,56 @@ def loadtest_template(
                     input_file_name, input_config.get('kind')))
 
             for client in input_config['spec']['clients']:
-                if client['language'] in client_languages:
-                    continue
+                del client['name']
                 if inject_client_pool:
                     client['pool'] = '${client_pool}'
-                clients.append(client)
-                client_languages.add(client['language'])
+                if client['language'] not in clientmap:
+                    clientmap[client['language']] = []
+                insert_worker(client, clientmap[client['language']])
 
             for server in input_config['spec']['servers']:
-                if server['language'] in server_languages:
-                    continue
+                del server['name']
                 if inject_server_pool:
                     server['pool'] = '${server_pool}'
-                servers.append(server)
-                server_languages.add(server['language'])
+                if server['language'] not in servermap:
+                    servermap[server['language']] = []
+                insert_worker(server, servermap[server['language']])
 
             input_spec = input_config['spec']
             del input_spec['clients']
             del input_spec['servers']
             del input_spec['scenariosJSON']
             spec.update(input_config['spec'])
 
-    clients.sort(key=lambda x: x['language'])
-    servers.sort(key=lambda x: x['language'])
+    uniquify_workers(clientmap)
+    uniquify_workers(servermap)
 
     spec.update({
-        'clients': clients,
-        'servers': servers,
+        'clients':
+            sum((clientmap[language] for language in sorted(clientmap)),
```

Comment:
That's a clever way of flattening a list of lists! FWIW, I don't think it's terribly efficient, but this is not performance-sensitive code.

Reply:
Thanks! It is more efficient than it looks, because it is just extending the list assigned to "start". I don't think you can do much better than that.